CMakeLists.txt

```diff
@@ -722,28 +724,32 @@ SET(targets)
 
 IF(PCRE2_BUILD_PCRE2_8)
   IF(BUILD_STATIC_LIBS)
-    ADD_LIBRARY(pcre2-8-static STATIC ${PCRE2_HEADERS} ${PCRE2_SOURCES} ${PROJECT_BINARY_DIR}/config.h)
+    ADD_LIBRARY(pcre2-8-static STATIC ${PCRE2_SOURCES})
     SET_TARGET_PROPERTIES(pcre2-8-static PROPERTIES
       COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8
       MACHO_COMPATIBILITY_VERSION "${LIBPCRE2_8_MACHO_COMPATIBILITY_VERSION}"
       MACHO_CURRENT_VERSION "${LIBPCRE2_8_MACHO_CURRENT_VERSION}"
       VERSION ${LIBPCRE2_8_VERSION}
       SOVERSION ${LIBPCRE2_8_SOVERSION})
     TARGET_COMPILE_DEFINITIONS(pcre2-8-static PUBLIC PCRE2_STATIC)
-    TARGET_INCLUDE_DIRECTORIES(pcre2-8-static PUBLIC ${PROJECT_BINARY_DIR})
+    TARGET_INCLUDE_DIRECTORIES(pcre2-8-static PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
```

Comment:
this might change the required minimum to use the library to something higher than 3.3.

it seems to work fine when building PCRE2 itself, but can't get it to work when using PCRE2 eventhough I might just not know how, and eitherway the old code doesn't work for me either, so it might not even be a regression

Reply:
[These generator expressions exist in 3.3.](https://cmake.org/cmake/help/v3.3/manual/cmake-generator-expressions.7.html)

Reply:
The generator expressions are safe. 

Comment 2:
However, the change from `PROJECT_BINARY_DIR` to `CMAKE_CURRENT_BINARY_DIR` is unrelated to this PR. 
The `${CMAKE_INSTALL_INCLUDEDIR}` works if no extra subdir must be added to care for the actual location below this prefix. (I didn't check yet.)

Reply:
Fixed.